dali/base/dadfs.cpp

```diff
@@ -8866,18 +8867,21 @@ class CFileScanner
             CFileMatch *match = &files.item(f);
             matchingFiles.append(*LINK(match));
             ++count;
+            if (checkFileCount && (count > ITERATE_FILTEREDFILES_LIMIT))
+                throw MakeStringException(-1, "More than %d files found, please use a filter to reduce the number of files returned.",
+                    ITERATE_FILTEREDFILES_LIMIT);
```

Comment:
That was my point, some sort of radix sort could produce the top N files based on date, size, name, etc. without hitting OOM error.  But do we wait for all files to come back or just limit that many coming back to keep response time decent ?